task/host-sp-comms/src/inventory/compute_sled_common.rs

```diff
@@ -170,41 +185,51 @@ impl crate::ServerImpl {
 fn read_fan_barcodes<T>(
     dev: I2cDevice,
     fans: &mut [T; 3],
+    barcode_buf: &mut [u8],
 ) -> Result<(), InventoryDataResult>
 where
     T: From<oxide_barcode::VpdIdentity>,
 {
     let [ref mut fan0, ref mut fan1, ref mut fan2] = fans;
-    *fan0 =
-        read_one_barcode::<VpdIdentity>(dev, &[(*b"SASY", 0), (*b"BARC", 1)])?
-            .into();
-    *fan1 =
-        read_one_barcode::<VpdIdentity>(dev, &[(*b"SASY", 0), (*b"BARC", 2)])?
-            .into();
-    *fan2 =
-        read_one_barcode::<VpdIdentity>(dev, &[(*b"SASY", 0), (*b"BARC", 3)])?
-            .into();
+    *fan0 = read_one_barcode::<VpdIdentity>(
+        dev,
+        &[(*b"SASY", 0), (*b"BARC", 1)],
+        barcode_buf,
+    )?
+    .into();
+    *fan1 = read_one_barcode::<VpdIdentity>(
+        dev,
+        &[(*b"SASY", 0), (*b"BARC", 2)],
+        barcode_buf,
+    )?
+    .into();
+    *fan2 = read_one_barcode::<VpdIdentity>(
+        dev,
+        &[(*b"SASY", 0), (*b"BARC", 3)],
+        barcode_buf,
+    )?
+    .into();
```

Comment:
we could probably make this use less stack by not doing the array-based structured representation and instead `memcpy`ing straight from `barcode_buf` into the inventory data fields...will do that if this isn't sufficient